src/NServiceBus.AcceptanceTests/DelayedDelivery/TimeoutManager/When_timeout_dispatch_fails_on_timeout_data_removal.cs

```diff
@@ -0,0 +1,167 @@
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading.Tasks;
+    using NServiceBus.AcceptanceTesting;
+    using NServiceBus.AcceptanceTests.EndpointTemplates;
+    using NServiceBus.AcceptanceTests.ScenarioDescriptors;
+    using NServiceBus.Extensibility;
+    using NServiceBus.Features;
+    using NServiceBus.Persistence;
+    using NServiceBus.Pipeline;
+    using NServiceBus.Timeout.Core;
+    using NUnit.Framework;
+
+    public class When_timeout_dispatch_fails_on_timeout_data_removal : NServiceBusAcceptanceTest
+    {
+        [Test]
+        public async Task Should_move_control_message_to_errors_and_not_dispatch_original_message_to_handler()
+        {
+            await Scenario.Define<Context>()
+                .WithEndpoint<Endpoint>(b => b.DoNotFailOnErrorMessages()
+                    .When((bus, c) =>
+                    {
+                        var options = new SendOptions();
+                        options.DelayDeliveryWith(TimeSpan.FromMinutes(1));
+                        options.RouteToLocalEndpointInstance();
+                        options.SetMessageId(c.TestRunId.ToString());
+
+                        return bus.Send(new MyMessage
+                        {
+                            Id = c.TestRunId
+                        }, options);
+                    }))
+                .Done(c => c.FailedTimeoutMovedToError)
+                .Repeat(r => r.For<AllTransportsWithoutNativeDeferral>())
+                .Should(c => Assert.IsFalse(c.DelayedMessageDeliveredToHandler, "Message was unexpectedly delivered to the handler"))
+                .Run();
+        }
+
+        public class Context : ScenarioContext
+        {
+            public bool FailedTimeoutMovedToError { get; set; }
+            public bool DelayedMessageDeliveredToHandler { get; set; }
+        }
+
+        public class Endpoint : EndpointConfigurationBuilder
+        {
+            public Endpoint()
+            {
+                EndpointSetup<DefaultServer>(config =>
+                {
+                    config.EnableFeature<TimeoutManager>();
+                    config.UsePersistence<FakeTimeoutPersistence>();
+                    config.SendFailedMessagesTo(AcceptanceTesting.Customization.Conventions.EndpointNamingConvention(typeof(Endpoint)));
+                    config.RegisterComponents(c => c.ConfigureComponent<FakeTimeoutStorage>(DependencyLifecycle.SingleInstance));
+                    config.Pipeline.Register<BehaviorThatLogsControlMessageDevliery.Registration>();
+                    config.LimitMessageProcessingConcurrencyTo(1);
+                });
+            }
+
+            class Handler : IHandleMessages<MyMessage>
+            {
+                public Context TestContext { get; set; }
+
+                public Task Handle(MyMessage message, IMessageHandlerContext context)
+                {
+                    if (message.Id == TestContext.TestRunId)
```

Comment:
Do we still need this? ( @timbussmann I remember that you had some plans to deal with it in a more generic way)